### PR TITLE
Fix/config

### PR DESCRIPTION
**Pull request by Sri-Harsha06**

### CODE REVIEW

File: generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/config/config_yaml.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package app
```
Comment (craxkumar) [inline generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/config/config_yaml.go]: Change the package name to the one which is set by end user

File: generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/config/config_yaml.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	"github.com/asim/go-micro/v3/config/reader"
+	"github.com/asim/go-micro/v3/config/reader/json"
+	"github.com/asim/go-micro/v3/config/source/file"
+	"fmt"
+	"github.com/micro/micro/v3/service/logger"
+)
+
+var g map[string]interface{}
+
+func Setconfig(){
```
Comment (craxkumar) [inline generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/config/config_yaml.go]: So this file contains the methods/logic to load the configuration based on the env variable passed, so let's name the file as configLoader.go rather then config_yaml.go

File: generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/dev-config.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+GO_MICRO_SERVICE_PORT : <%= serverPort %>
```
Comment (craxkumar) [inline generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/dev-config.yaml]: Let's keep all the yaml files in one palce, all the .yml files can go in the config dir.  For good user readability.

**Review by craxkumar — CHANGES_REQUESTED** [review]
We would required .env file for the mainfest files to work, 
Let's discuss tomorrow!

File: generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/config/config_yaml.go (hunk 1 above)
Comment (Sri-Harsha06) [reply generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/config/config_yaml.go]: This is like a module not the global package name as we are not taking this as input from the user it is predefined.

File: generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/config/configLoader.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	yaml "github.com/asim/go-micro/plugins/config/encoder/yaml/v3"
+	"github.com/asim/go-micro/v3/config/reader"
+	"github.com/asim/go-micro/v3/config/reader/json"
+	"github.com/asim/go-micro/v3/config/source/file"
+	"fmt"
+	"strings"
+	"github.com/micro/micro/v3/service/logger"
+)
+
+var g map[string]interface{}
```
Comment (craxkumar) [inline generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/config/configLoader.go]: change it to some useful name

File: generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/config/configLoader.go [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+func Setconfig(){
+	profile := "dev"
+	profiles := map[string]bool{"dev":true, "prod":true};
+	if( profiles[os.Getenv("GO_MICRO_PROFILE")] ){
+		profile = os.Getenv("GO_MICRO_PROFILE");
+	}
+	enc := yaml.NewEncoder()
+	c, _ := config.NewConfig(
```
Comment (craxkumar) [inline generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/config/configLoader.go]: change it to meaning full name

File: generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/eurekaregistry/EurekaRegistrationManager.go [hunk 6]
```diff
@@ -99,33 +78,27 @@ func (erm EurekaRegistrationManager) DeRegisterFromServiceRegistry(configs Regis
...
-	// if err != nil{
-	// 	logger.Errorf("error","Enable to find hostname form OS, sending appname as host name")    	
-	// }
-	hostname := "<%= baseName %>"
+	httpport := app.GetVal("GO_MICRO_SERVICE_PORT")
+	hostname, _ := os.Hostname()
+	
+	env :=os.Getenv("GO_MICRO_PROFILE")
+	if(env=="prod"){
+		//set this as hostname for local environment
```
Comment (craxkumar) [inline generator-jhipster-gomicro/generators/server/templates/gomicro/gomicro/eurekaregistry/EurekaRegistrationManager.go]: remove this comment

**Review by craxkumar — CHANGES_REQUESTED** [review]

**Review by craxkumar — APPROVED** [review]
Review OK!

**Pull request merged**